### ISSUE TITLE
Add Version to relationship aspects

### ISSUE THREAD

**Issue by jzbmw** [eclipse-tractusx/item-relationship-service#470]
**As** PO ,
**I want** to be compliant to the versioning initiative of CX ,
**so that** we will be downward compatible .

## Link 
- child of  https://github.com/eclipse-tractusx/sig-release/issues/650
- tested with https://github.com/eclipse-tractusx/item-relationship-service/issues/529

## Hints / Details
- example Value for a bomlifecycle element 
- it shall be possible, that two versions of the same aspect can exist in parallel of a bomlifecycle
- introduce a list of concrete semantic urn's urn:samm:io.catenax.single_level_bom_as_built:3.0.0

## Assumption 
As the semantic models are using semantic versioning, the api and in this case the semantic model will not provide braking changes with minor or patch version updates. 
Instead of supporting only a specific version for traversal aspects which cause sirs updates on each semantic model patch version update, IRS should be made resilience to support a major version of semantic models 

## Acceptance Criteria
- [x] IRS supports the latest full major version range for each traversal aspect (3.0.0 - 3.10.0) 
- [x] All known versions of current major versions of traversal aspects SingleLevel*As* (3.0.0) are covered with tests 
- [x] the API Documentation has been updated
- [x] Tests have been updated
- [ ] ~~The API has been changed to accept only the urn's which are available~~
- [x] Add valid example values to API docu ( replace "string")
- [x] Adjusting tests 
- [x] Adjusting mocks 
- [x] Adjusting API documentation 
- [x] Adjusting documentation 

## General 
* aspectType is provided (Trace-X)
* linkedItem is still used 

### SingleLevelBomAsBuilt | 3.0.0 |  
* catenaXId (Trace-X)
* childItems/catenaXId (Trace-X)
* hasAlternatives
* businessPartner (Trace-X)

### SingleLevelBomAsPlanned | 3.0.0
* catenaXId (Trace-X)
* childItems/catenaXId (Trace-X)
* businessPartner (Trace-X)

### SingleLevelUsageAsBuilt | 3.0.0
* catenaXId (Trace-X)
* parentItems/catenaXId (Trace-X)
* parentItems/isOnlyPotentialParent
* businessPartner (Trace-X)

### SingleLevelUsageAsPlanned | 2.0.0
* catenaXId (Trace-X)
* parentItems/catenaXId (Trace-X)
* businessPartner (Trace-X)

### SingleLevelBomAsSpecified | 2.0.0 
* assetId
* childItems/childassetId

## Out of Scope

- ...

**Comment by mkanal** [comment]
# Discussion
Idea is to use the major version similar to the approach of semantic models. 
We keep this open as we want to reevaluate this idea with  @ds-jhartmann

**Comment by mkanal** [comment]
Label was removed as there is a clear decision to support a full major version range for each traversal aspects. Currently, 3.0.0 - 3.x.x

**Comment by ds-jhartmann** [comment]
Implemented in https://github.com/eclipse-tractusx/item-relationship-service/pull/671

- Support for SingleLevelUsageAsPlanned.
- Documentation to describe the delegate process.
- Added Tests for previous supported SingleLevel* Versions
- Added Tests for potential future minor/patch SingleLevel* Versions

As discussed wiht @mkanal, the AC "The API has been changed to accept only the urn's which are available" was not implemented, since the relationship aspects are not directly specified via API, only inferred by "bomLifecycle" and "direction".

**Comment by ds-jhartmann** [comment]
This story will be tested by #529

**Comment by mkanal** [comment]
AC: "The API has been changed to accept only the urn's which are available" Decision: AC was removed as the urn will not be published over the api.
